### ISSUE TITLE
Hardware PWM frequency on ATMega 2560 too low

### ISSUE THREAD

**Issue by philOmatic321** [KevinOConnor/klipper#4224]
**TL;DR:** frequency of hardwarePWM only slightly higher than softwarePWM and both are too low for my part cooling fan

I switched my ATMega 2560 based Trigorilla mainboard from Marlin to Klipper and everything works surprisingly well. Except for my part cooling fan. I use a Sunon EF50151B1 and at 100% it works fine. As soon as I lower the duty cycle it starts to behave very weird, even a `M106 S254` will make it ramp up and down. every value lower than that will shut it completely off. 

After controlling the signal with an oscilloscope, the signal is fine but the softwarePWM frequency of 100Hz is rather low. When connecting a 100µF electrolytic capacitor in parallel to the fan it behaves normally. I therefore suspect that it would run with the default hardware configuration if the PWM frequency would be higher.

I therefore switched over to hardwarePWM. As the fan is connected to Pin D9 this should be possible with Timer2 on the ATMega. After measuring the frequency again it increased to 122Hz which is way lower than what I expected and obviously doesn't make a difference for my fan. 

Is Timer2 used by the Klipper firmware? If not, would it be possible to increase the hardwarePWM frequency to something at least 10 times the softwarePWM frequency? 

Thanks for your help!

You can find the logfile here:
[klippy.log](https://github.com/KevinOConnor/klipper/files/6382565/klippy.log)
